### PR TITLE
Add a rule to bump controller-gen tool in Dockerfiles

### PR DESCRIPTION
**Pull request by maxsokolovsky**

### CODE REVIEW

**Review by macedogm — DISMISSED** [review]

File: default.json [hunk 1]
```diff
@@ -80,6 +80,17 @@
       "depNameTemplate": "mikefarah/yq",
       "datasourceTemplate": "github-releases"
     },
+    {
+      "fileMatch": [
+        "(^|/|\\.)Dockerfile$",
+        "(^|/)Dockerfile[^/]*$"
```
Comment (pjbgf) [inline default.json:86]: Let's expand this to Makefiles as well please:
- https://github.com/rancher/elemental-operator/blob/80eb8469e31cb0b88923a0a2e26713e88928e4dc/Makefile#L48
- https://github.com/rancher-sandbox/cluster-api-provider-rke2/blob/e05b03eb7a3cc664ae2f75a11741d7b7cb0a3d53/Makefile#L73

File: default.json [hunk 2]
```diff
@@ -80,6 +80,17 @@
...
+      ],
+      "matchStrings": [
+        "ENV CONTROLLER_GEN_VERSION(\\=|\\s)(?<currentValue>.*?)\\n"
```
Comment (pjbgf) [inline default.json:89]: We could add an alternative pattern for: `CONTROLLER_GEN_VER := ...` as per Makefile examples above.

**Review by KevinJoiner — APPROVED** [review]
It looks safe to have automatic bumps to controller gen

File: default.json (hunk 2 above)
Comment (maxsokolovsky) [reply default.json:89]: We'd need to first change the name of the ENV in the respective Dockerfile, or could we just make the pattern shorter here?

Comment (pjbgf) [reply default.json:89]: Either way really, we can either change in the repositories to use a single pattern, or add both patterns here until that take place. I can see 2 patterns across 4 files:

image

**Review by macedogm — APPROVED** [review]

**Review by pjbgf — APPROVED** [review]

**Pull request merged**